File: core/src/main/java/com/ortin/gesturetranslator/core/managers/mediapipe/MediaPipeManager.java
package com.ortin.gesturetranslator.core.managers.mediapipe;

import com.ortin.gesturetranslator.core.managers.mediapipe.listeners.MPDetectionListener;
import com.ortin.gesturetranslator.core.managers.mediapipe.models.MPImageInput;

public interface MediaPipeManager {
    void detectImage(MPImageInput mpImageInput);

    void setMPDetectionListener(MPDetectionListener mpDetectionListener);
}

File: core/src/main/kotlin/com/ortin/gesturetranslator/core/managers/mediapipe/MediaPipeManager.kt
package com.ortin.gesturetranslator.core.managers.mediapipe

import android.graphics.Bitmap
import com.ortin.gesturetranslator.core.managers.mediapipe.listeners.MPDetectionListener
import com.ortin.gesturetranslator.core.managers.mediapipe.models.MPImageDetection
import com.ortin.gesturetranslator.core.managers.mediapipe.models.MPVideoDetection
import com.ortin.gesturetranslator.core.managers.mediapipe.models.MPVideoInput
import com.ortin.gesturetranslator.core.managers.mediapipe.models.SettingsModel

interface MediaPipeManager {
    fun detectImage(image: Bitmap): MPImageDetection?
    fun detectVideoFile(videoFile: MPVideoInput): MPVideoDetection?
    fun detectLiveStream(image: Bitmap)
    fun setSettingsModel(settingsModel: SettingsModel)
    fun setMPDetectionListener(mpDetectionListener: MPDetectionListener)
}
